Enso example development

sana-ccrc opened the pull request:
Changes made to **ENSO_Forecast** and **ENSO_Pipeline** notebooks:
- Added a script that prepares a dataframe ready for machine learning, including target variables needed for 1-month lag forecasts with XGBoost and 3-month lag forecasts with MLP
- Added a script that encodes 'month' into cyclic features; this is useful if we need to include month as a predictor, using 'month_sin' and 'month_cos'
- minor: edited subheadings and cleaned comments

_Issue: I am using pet/2025.06, and there is no **plotly** library_

sana-ccrc committed:
Update the introduction and added references and a reference section.

tennlee committed:
Add example of connecting to ERA5 and extracting a bounding box region from the monthly averaged data

sana-ccrc committed:
Merge pull request #1 from tennlee/sanaa_enso

Addition of connecting to ERA5 monthly averaged data and extracting a bounding box

sana-ccrc committed:
Merge remote-tracking branch 'origin/develop' into enso_example_development

sana-ccrc committed:
Add a script that calculates nino3.4 index to ENSO_Forecast.ipynb and delete ENSO_Prediction.ipynb

tennlee committed:
Demonstrate the use of the pipeline for:
 - Accessing the bounding box
 - Calculating the mean
 - Conversion of the pipeline data into an xarray object

sana-ccrc committed:
changes made to ENSO_Forecast and ENSO_Pipeline:
- Added a script that prepares a dataframe ready for machine learning, including target variables needed for 1-month lag forecasts with XGBoost and 3-month lag forecasts with MLP
- Added a script that encodes 'month' into cyclic features; this is useful if we need to include month as a predictor, using 'month_sin' and 'month_cos'
- minor: edited subheadings and cleaned comments

Issue: I am using pet/2025.06, and there is no plotly library

sana-ccrc committed:
Merged with origin/enso_example_development, kept local changes for ENSO_Forecast and ENSO_Pipeline

tennlee commented:
Thanks, I've had a read through of this. Let's have a chat early next week about making the ML objectives jump out just a little more, set up the XGB model and work up a good test/train/validate split etc.

sana-ccrc commented:
Sounds good. Are you available on Tuesday 29 July anytime between 12 and 5pm?

sana-ccrc committed:
Added background and preprocessed data; ready for XGBoost training

sana-ccrc commented:
Added more background
Preprocessed the data further, now it is ready for XGBoost training

sana-ccrc committed:
Added a column with the monthly climatology for nino3.4 for comparison with the ML model predictions.

sana-ccrc commented:
Computed monthly climatology of Nino3.4, which can be used as a benchmark to evaluate our model.

sana-ccrc committed:
Update ENSO_Forecast.ipynb

tennlee committed:
Add inline comments in ENSO pipeline for review
Add gridded MLP example (some more commentary and evaluation to be added but well on the way)

tennlee commented:
(note - the unit tests have started failing. This is completely unrelated to this change, and pertains to how the version is set in the nci site archive subpackage, and I think it's a change in strictness in the tooling. I will fix it separately, ignore it)

tennlee committed:
Add gridded MLP notebook

sana-ccrc committed:
Tidy up ENSO_Forecast.ipynb: fix layout and remove unused cells

sana-ccrc committed:
Add ENSO Tutorial folder with notebooks

sana-ccrc committed:
Update ENSO_Forecast notebook

tennlee committed:
Now that the tutorials have been moved into a dedicated folder and tidied up, remove the working copies

tennlee commented:
@sana-ccrc Thanks so much for the work you did here. I have merged this pull request into PyEarthTools. There were duplicate copies of the notebooks which I removed, feel free to confirm I did the right thing and I can restore the deleted notebooks easily if needed. I will also go ahead and link these tutorials into the documentation so that they can be found easily, and I'll tweak the gridded MLP notebook as well as we discussed.

In addition, consider whether you would like to be listed in the .zenodo.json for the repository.

When we release a new version of PyEarthTools, that version is archived on Zenodo. See: https://doi.org/10.5281/zenodo.15760768

As you have contributed to PyEarthtools, would you like to be listed on Zenodo as an author the next time PyEarthTools is archived?

If so, please open a new pull request. In that pull request please add your details to .zenodo.json (which can be found in the PyEarthTools root directory).

In .zenodo.json, please add your details at the bottom of the “creators” section. The fields you will need to complete are:

1. “orcid”. This is an optional field. If you don’t have an ORCID, but would like one, you can obtain one here: https://info.orcid.org/researchers/ .
2. “affiliation”. Options include: the institution you are affiliated with, “Independent Researcher” or “Independent Contributor”.
3. “name”. Format: surname, given name(s).
